### PR TITLE
builtins: Implement ST_FlipCoordinates on arguments {geometry}

### LINKED ISSUE #48932: geo/geomfn: implement ST_FlipCoordinates({geometry})
Implement `ST_FlipCoordinates` on arguments {geometry}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_FlipCoordinates.html).

_Observers: Please react to this issue if you need this functionality._

For Geometry builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geomfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geomfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
  * When using GEOS, you can reference the [C API](https://github.com/libgeos/geos/blob/master/capi/geos_c.h.in) for which functions are available. Unfortunately, Windows is not currently supported when using GEOS.
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end (your exhaustive unit tests go the above mentioned packages!).
* Ensure the documentation is regenerated by calling `make buildshort`. You can also play with it by calling `./cockroach demo --empty` afterwards.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48552 for an example PR.

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2020-06-30T05:29:57Z. Changes to titles, body and labels may be overwritten.</sub>

### PR DESCRIPTION
**Pull request by themistoklik**
Should adopt PostGIS behavior. 

Release note (sql change): Implemented geometry builtin `ST_FlipCoordinates`
Closes #48932

### CODE REVIEW

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 1]
```diff
@@ -3708,6 +3708,28 @@ Bottom Left.`,
...
+
+				ret, err := geomfn.FlipCoordinates(g.Geometry)
+				if err != nil {
+					return nil, err
+				}
+
+				return tree.NewDGeometry(ret), nil
+			},
+			Info: infoBuilder{
+				info: "Returns a new geometry with X,Y axes flipped.",
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:3728]: nit: can you make this `Returns a new geometry with the X and Y axes flipped.`

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 2]
```diff
@@ -3708,6 +3708,28 @@ Bottom Left.`,
 			tree.VolatilityImmutable,
 		),
 	),
+	"st_flipcoordinates": makeBuiltin(
+		defProps(),
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"geometry", types.Geometry}},
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:3715]: nit: new line per argument for multiline args, i.e.

```
Types: tree.ArgTypes{
				{"geometry", types.Geometry},
},
```

File: pkg/geo/geomfn/flip_coordinates.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+	return g, nil
+}
+
+// flipCoords swaps X,Y coordinates.
+// M and Z coordinates are maintained
+func flipCoords(g geom.T) ([]float64, error) {
+	stride := g.Stride()
+	coords := g.FlatCoords()
+	newCoords := make([]float64, len(coords))
```
Comment (otan) [inline pkg/geo/geomfn/flip_coordinates.go:72]: any reason to not re-use the same array here?

**Review by otan — COMMENTED** [review]
Hey! Thanks for going through this! Some minor comments.

File: pkg/sql/sem/builtins/geo_builtins.go (hunk 1 above)
Comment (themistoklik) [reply pkg/sql/sem/builtins/geo_builtins.go]: Done.

File: pkg/sql/sem/builtins/geo_builtins.go (hunk 2 above)
Comment (themistoklik) [reply pkg/sql/sem/builtins/geo_builtins.go]: Done.

File: pkg/geo/geomfn/flip_coordinates.go (hunk 3 above)
Comment (themistoklik) [reply pkg/geo/geomfn/flip_coordinates.go]: Done. Good point, removed the M/Z copying too.

**Review by otan — APPROVED** [review]

**Comment by SWDevAngel** [comment]
@themistoklik Hi Themis. Thank you for contributing to CockroachDB this year. As a token of our appreciation we would like to send you a gift. Please DM me on our [community Slack](https://cockroa.ch/slack) `@lisa` so I can send you a link. (If you don’t want a gift, we also have a charitable contribution choice.) All orders need to be in by December 13, so please send this asap. :)

**Pull request merged**